torch_geometric/sampler/neighbor_sampler.py

```diff
@@ -249,9 +280,71 @@ def sample(
             raise TypeError(f'{self.__class__.__name__} found invalid type: '
                             f'{type(self.data_cls)}')
 
+    # Edge-based sampling #####################################################
 
-###############################################################################
+    def sample_from_edges(
+        self,
+        index: EdgeSamplerInput,
+        **kwargs,
+    ) -> Union[SamplerOutput, HeteroSamplerOutput]:
```

Comment:
```suggestion
    ) -> Union[SamplerOutput, HeteroSamplerOutput]:
    r"""Implements neighbor sampling starting from edges in index, 
    by calling 'torch-sparse' sampling routines, conditional on the type of data object."""
```

Reply:
Added this (and the above comment) in the base class.